crates/adapters/src/synchronizer/mod.rs

```diff
@@ -0,0 +1,180 @@
+mod error;
+mod rclone;
+
+use anyhow::Result;
+use dbsp::circuit::checkpointer::{CheckpointMetadata, CHECKPOINT_FILE_NAME};
+use error::SyncError;
+use feldera_adapterlib::errors::journal::ControllerError;
+use feldera_types::config::SyncConfig;
+use rclone::{RcloneLocation, RcloneSync};
+use serde::Deserialize;
+use std::{
+    ffi::OsStr,
+    fs::File,
+    io::BufReader,
+    path::{Path, PathBuf},
+    process::ExitStatus,
+};
+use tokio::io::AsyncReadExt;
+
+use crate::controller::STATE_FILE;
+
+#[derive(Debug, Deserialize)]
+struct PSpineFiles {
+    files: Vec<String>,
+}
```

Comment:
This duplicates PSpineBatches in crates/dbsp/src/trace/spine_async/mod.rs. I think it would be better to use it directly. (If it's not `pub` enough then you can export it more.)